Assets/Scripts/Internal/DaggerfallMobileBillboard.cs

```diff
@@ -0,0 +1,734 @@
+// Project:         Daggerfall Tools For Unity
+// Copyright:       Copyright (C) 2009-2020 Daggerfall Workshop
+// Web Site:        http://www.dfworkshop.net
+// License:         MIT License (http://www.opensource.org/licenses/mit-license.php)
+// Source Code:     https://github.com/Interkarma/daggerfall-unity
+// Original Author: Gavin Clayton (interkarma@dfworkshop.net)
+// Contributors:    TheLacus
+// 
+// Notes:
+//
+
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+using System;
+using System.IO;
+using DaggerfallConnect;
+using DaggerfallConnect.Utility;
+using DaggerfallConnect.Arena2;
+using DaggerfallWorkshop.Utility;
+using DaggerfallWorkshop.Utility.AssetInjection;
+using DaggerfallWorkshop.Game.Utility;
+
+namespace DaggerfallWorkshop
+{
+    /// <summary>
+    /// A billboard class for classic Daggerfall mobile sprites with 8 orientations.
+    /// Handles loading resources and rendering billboard based on orientation and state.
+    /// Will rotate and scale based on camera angle and texture, so this component should
+    /// only be attached to a child GameObject.
+    /// Uses parent GameObject to determine actual facing in world.
+    /// </summary>
+    #if UNITY_EDITOR
+    [ExecuteInEditMode]
+    #endif
+    [RequireComponent(typeof(MeshFilter))]
+    [RequireComponent(typeof(MeshRenderer))]
+    public class DaggerfallMobileBillboard : DaggerfallMobileUnit
+    {
+        const int numberOrientations = 8;
+        const float anglePerOrientation = 360f / numberOrientations;
+
+        [SerializeField]
+        MobileUnitSummary summary = new MobileUnitSummary();
+
+        Camera mainCamera = null;
+        MeshFilter meshFilter = null;
+        MeshRenderer meshRenderer = null;
+
+        Vector3 cameraPosition;
+        float enemyFacingAngle;
+        int lastOrientation;
+        int currentFrame;
+        int frameIterator;
+        bool restartAnims = true;
+
+        public sealed override bool IsSetup
+        {
+            get { return summary.IsSetup; }
+        }
+
+        public sealed override MobileEnemy Enemy
+        {
+            get { return summary.Enemy; }
+        }
+
+        public sealed override MobileStates EnemyState
+        {
+            get { return summary.EnemyState; }
+        }
+
+        public sealed override int AnimStateRecord
+        {
+            get { return summary.AnimStateRecord; }
+        }
```

Comment:
Used only here:

```cs
int damage = FormulaHelper.CalculateAttackDamage(playerEntity, enemyEntity, entityMobileUnit.AnimStateRecord, animTime, strikingWeapon);
```

I understand is used to detect backstabbing:

```cs
// If enemy is facing away from player
if (enemyAnimStateRecord % 5 > 2)
{
    player.TallySkill(DFCareer.Skills.Backstabbing, 1);
    return player.Skills.GetLiveSkillValue(DFCareer.Skills.Backstabbing);
}
```

Can i replace it with `bool IsBackFacing`?

Reply:
That would make a much better interface IMO. so +1 from me